### PR TITLE
Improved docs, consolidated release script, and faster builds

### PR DESCRIPTION
**Pull request by mgasner**

### CODE REVIEW

File: python_modules/dagster/docs/installation.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
+Installation
+=======================
+
+Dagster is tested on Python 3.6.6, 3.5.6, and 2.7.15. Python 3 is strongly
+encouraged -- if you can, you won't regret making the switch!
```
Comment (schrockn) [inline python_modules/dagster/docs/installation.rst]: 👍🏻

File: python_modules/dagster/docs/installation.rst [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+`install yarn <https://yarnpkg.com/lang/en/docs/install/>`_.
+
+Creating a virtual environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+We strongly recommend installing dagster inside a Python virtualenv. If you are
+running Anaconda, you should install dagster inside a Conda environment.
+
+To create a virtual environment on Python 3, you can just run:
+::
+    $ python3 -m venv /path/to/new/virtual/environment
```
Comment (schrockn) [inline python_modules/dagster/docs/installation.rst]: aside: we may want to consider switching to and recommending pipenv at some point. they have lockfiles!

File: bin/publish.py [hunk 3]
```diff
@@ -0,0 +1,284 @@
+import contextlib
+import os
+import re
+import subprocess
+
+import click
+import packaging.version
+
+from itertools import groupby
+
```
Comment (schrockn) [inline bin/publish.py]: This is script is definitely worthy of a nice long comment that explains what is going on here. From what I read it is searching for a git tag, overriding the version file, commiting it and publishing?

**Comment by mgasner** [comment]
Oh yeah I was going to do that "later". Added a bunch of documentation on the command line.

**Pull request merged**